### PR TITLE
docs: update to simplified threads spec

### PR DESCRIPTION
**Pull request by oed**
This updates the threads spec to be much simpler for the first iteration. Basically we only allow append only threads without any moderation.

### CODE REVIEW

File: 3IPs/3ip-2.md [hunk 1]
```diff
@@ -9,92 +9,68 @@ created: 2018-11-20
...
-const thread = await box.threads.new(<name>, <open=false>)
+const threadNames = space.threads.list()
+console.log(threadNames)
+> [{ name: "3box-chat", open: true }, { name: "Bounties-issue432", open: false }]
 ```
 
-**Get thread id**
-
-the thread-id is simply the orbitdb-address (String) of the underlaying DB.
+**Create an existing thread:**
```
Comment (michaelsena) [inline 3IPs/3ip-2.md]: Should this be: 'Follow/Subscribe to an Existing Thread'  ?

File: 3IPs/3ip-2.md [hunk 2]
```diff
@@ -9,92 +9,68 @@ created: 2018-11-20
...
+const thread = await threads.threads.open(<name>)
 ```
 
-**Subscribe to an existing thread:**
+**Create a new thread:**
+When a new thread is created its orbitdb address is also added to the users space under the key `thread-<thread-name>`.
 
 ```js
-const thread = await box.threads.subscribe(<thread-id>, <name>) 
+const thread = await threads.threads.new(<name>)
```
Comment (michaelsena) [inline 3IPs/3ip-2.md]: is this weird to have threads.threads  ?

File: 3IPs/3ip-2.md [hunk 3]
```diff
@@ -9,92 +9,68 @@ created: 2018-11-20
...
-const threadNames = box.threads.list()
-console.log(threadNames)
-> ["3box-chat", "Bounties-issue432"]
+const space = box.openSpace('aGoodSpace')
 ```
 
-**Create a new thread:**
-
-The `open` parameter is a boolean that specifies if there is write permissions to the thread or not. Open threads anyone can post to, but there is no spam protection. Note that new threads are not automatically subscribed to.
+**List threads the user has interacted with**
```
Comment (michaelsena) [inline 3IPs/3ip-2.md]: Should this be: "List Threads a User is Following/Subscribed to"  ?

File: 3IPs/3ip-2.md [hunk 4]
```diff
@@ -9,92 +9,68 @@ created: 2018-11-20
...
-
-## Thread subscriptions
-
-Users can subscribe to threads. This means that they will remember the thread next time they go online, and they can ask the network for updates. When a user subscribes to a thread it simply adds the thread-id and name to the rootStore of the users 3Box. When the user unsubscribes the thread entry is removed from the rootStore.
-
-## Threads in spaces?
-
-Is this something we want to enable? Maybe as a second iteration, it adds a lot of complexity.
+## Thread creation
+Threads are created inside of [Spaces](./3ip-1.md). A thread has a `name` which could be any string. For example the name could be the address of a contract if you want to create a thread for commenting on that contract. This also means that two different spaces could have their own threads with the same name. When a thread is created it gets the name `3box.thread.<space-name>.<thread-name>` for the orbitdb logstore.
```
Comment (michaelsena) [inline 3IPs/3ip-2.md]: What does it mean for a thread to be inside of a space? If the subject of a thread is the address of a contract, for example, would this thread live in the 3Box of the app address that enabled the UI to take place? Or does the root thread live in the user's app space?

File: 3IPs/3ip-2.md (hunk 2 above)
Comment (oed) [reply 3IPs/3ip-2.md]: oh, should be `space` of course :)

File: 3IPs/3ip-2.md (hunk 3 above)
Comment (oed) [reply 3IPs/3ip-2.md]: It could be, do you think we should add a method for unsubscribing?

Comment (michaelsena) [reply 3IPs/3ip-2.md]: I think it couldn't hurt, and could allow someone to be very efficient in the threads they follow. Makes logical sense.

File: 3IPs/3ip-2.md (hunk 4 above)
Comment (oed) [reply 3IPs/3ip-2.md]: It means that the access control for writing to the thread is based on the space. Say space `A` and space `B` are used by two different apps, but they both comment on contract `X`. This means that there will be two separate threads, and also that only dapps which the user has allowed to open space `A` can open and write to the threads in that space.

I'm not exactly sure what you mean by "root thread"? All threads are essentially the same.

File: 3IPs/3ip-2.md (hunk 1 above)
Comment (oed) [reply 3IPs/3ip-2.md]: oh, It should be "Open an existing thread"

Comment (oed) [reply 3IPs/3ip-2.md]: I'm wondering if we should make a distinction between opening and following a thread though.

**Comment by oed** [comment]
@michaelsena I think the user flow is a bit out of scope for this 3ip.

File: 3IPs/3ip-2.md (hunk 1 above)
Comment (michaelsena) [reply 3IPs/3ip-2.md]: I don’t know what the difference is. Can you pls explain?

Comment (oed) [reply 3IPs/3ip-2.md]: You might want to open a thread to view it without actually following it (across multiple sessions).

**Comment by michaelsena** [comment]
@oed I still think it could be helpful for you to write out a quick user/dev flow, so i can see how user actions translate to api methods. It will demonstrate the use case for each method so we all can better understand how the system works.

File: 3IPs/3ip-2.md (hunk 4 above)
Comment (zachferland) [reply 3IPs/3ip-2.md]: How does this hold `only dapps which the user has allowed to open space A can open and write to the threads in that space`? is there some values from opening the space that derive a different db manifest

Comment (oed) [reply 3IPs/3ip-2.md]: @zachferland only that the name of the odb store is `3box.thread.<space-name>.<thread-name>`

Comment (oed) [reply 3IPs/3ip-2.md]: @zachferland only that the name of the odb store is `3box.thread.<space-name>.<thread-name>`

Comment (zachferland) [reply 3IPs/3ip-2.md]: so basically if entire name is known, it can be written to?

Comment (oed) [reply 3IPs/3ip-2.md]: exactly yea :)

**Pull request merged**